Backpack-SwiftUI/SearchInputSummary/Classes/BPKSearchInputSummary.swift

```diff
@@ -0,0 +1,136 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2022 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+/// A control that displays an editable text interface for search inputs.
+///
+/// Use `inputState(_ state: State)` to change the state of the text field.
+public struct BPKSearchInputSummary: View {
+    public enum InputPrefix {
+        case text(String)
+        case icon
```

Comment:
this should receive a BPKIcon as parameter, so that clients can define what icon they use (clients of this component may be within backpack, but at least it would be configurable from outside this very component)

Reply:
hmm, that's a good idea I guess to make it more flexible